### ISSUE TITLE
IGVjs & Nginx unable to properly set symlinks to correct datasets

### ISSUE THREAD

**Issue by florianzwagemaker** [DennisSchmitz/Jovian_archive#25]
The symlink to igvjs is made with the script `bin/start_nginx.sh` with the if-statement underneath
```
   # make link to igv software
   if [ ! -h $CONDA_PREFIX/etc/nginx/default-site/igv ]
   then
      ln -s "$(pwd)/bin/software/igv.js" "$CONDA_PREFIX/etc/nginx/default-site/igv"
   fi
```
This creates 2 possible problems:

1. The symlink cannot be set when nginx is already running since the script `bin/start_nginx.sh` exits when nginx is running.
This also is a problem when nginx hasn't exited properly. It will become impossible to (re)start or stop nginx with the therefore intended script because nginx will keep running without a valid PID (and pid file, this is most likely because the nginx process is started through jupyter). The only workaround for this is to completely delete the `Jovian_master` environment.

2. The symlink is generated with a generic name: "`$CONDA_PREFIX/etc/nginx/default-site/igv`"
Because the symlink is working with a generic **igv** name the symlink has to be overwritten each time you wish to view a new report/dataset.
This makes it nearly impossible to *properly* view multiple reports/datasets at the same time. Especially when problem 1 is in effect at the same time.

- [x] Should investigate if the `bin/start_nginx.sh` script can be rewritten in order to set the symlink even when nginx itself isn't running.
- [x] Make use of **random** symlink names in order to support multiple datasets at the same time, generating a random number could be implemented into the wrapper --> #7 
- [x] Call the starting of the nginx server process through the wrapper (#7) to make sure nginx starts and stops properly.

Random numbers could be written/read in the same way as #20 could be implemented (writing to, and parsing/reading from yaml files)

**Comment by florianzwagemaker** [comment]
Update 1;
Generated symlinks are now a random combination of letters and numbers, the random strings have a length of 32 characters.

the script `start_nginx.sh` no longer sets the symlink (from commit https://github.com/DennisSchmitz/Jovian/commit/eabde9b3b40b6dd1d71e05fe9966cc3d7e2aa71c)

The randomized symlink is now set by the snakefile on completion of the pipeline (OnSuccess statement)

IGV rule in snakefile (create index) has been changed and redone in a separate shell script in order to make IGVjs' html files work with the random string.

**Comment by florianzwagemaker** [comment]
Moved milestone to 0.9.3 for now.
This specific issue itself will be solved completely for 0.9.3.
0.9.2 will however feature multiple fixes to the various problems that are listed in this issue, such as random identifiers, random symlinks and modified scripts for nginx.

0.9.3 will further improve and develop these features aswell as better starting/stopping of the nginx process (and hopefully Jupyter aswell)
